apps/frontend/src/modules/surveys/SurveyProvider.tsx

```diff
@@ -0,0 +1,609 @@
+import {
+    OrderDirection,
+    PageInfo,
+    ProjectTheme,
+    Survey,
+    SurveyCountableConnection,
+    SurveyError,
+    SurveyFilterInput,
+    SurveyQuestionCountableEdge,
+    SurveyQuestionTypeEnum,
+    SurveySortField,
+    SurveyStatusEnum,
+    SurveyTypeEnum,
+    SurveyUpdateInput,
+    useGetSurveyListLazyQuery,
+    useGetSurveyQuery,
+    useSurveyCreateMutation,
+    useSurveyDeleteMutation,
+    useSurveyQuestionCreateMutation,
+    useSurveyUpdateMutation,
+} from "@/generated/graphql";
+import { ROUTES } from "@/routes";
+import { generateRandomString } from "@/utils";
+import { createSelectors } from "@/utils/selectors";
+import { ApolloError, InMemoryCache, Reference } from "@apollo/client";
+import { relayStylePagination } from "@apollo/client/utilities";
+import React from "react";
+import { useNavigate, useParams } from "react-router-dom";
+import { useScrollToBottom } from "react-scroll-to-bottom";
+import useWorkspace from "../workspace/hooks/useWorkspace";
+import { SURVEY, SURVEY_QUESTION } from "./graphql/fragments/surveyFragment";
+import { useSurveyStore } from "./states/survey";
+
+export interface SurveyProviderProp {
+    children: React.ReactNode;
+}
+
+export type SurveyResponse = {
+    surveyErrors: SurveyError[] | null;
+    survey?: Survey | null;
+};
+
+export type SurveyContextValues = {
+    surveysOnPage: number;
+    totalSurveys: number | null | undefined;
+    pageInfo: PageInfo | undefined;
+    loading: boolean;
+    error: ApolloError | undefined;
+    openQuestion: string;
+    surveyName: string | undefined;
+    selectedSurveyTheme: string | undefined;
+    surveyId: string | undefined;
```

Comment:
@kaf-lamed-beyt Why do we have all these variables? I thought they are all already in `survey`. Can you remove `surveyId`, `surveyName`, `selectedSurveyTheme`, `pageInfo` e.t.c. They are alternative ways to access these values.